src/js/components/MosaicView.js

```diff
@@ -204,3 +268,43 @@ ChannelMetadata.propTypes = {
 
 const buildMediaURI = (locationName, cameraName, channelName, filename) =>
   `${locationName}/${cameraName}/${channelName}/${filename}`
+
+function frameStep(vidID, timeDelta) {
+  console.log('frame delta is: ',timeDelta)
+  const video = _getById(vidID)
+  pauseVideo(video)
+  if (timeDelta < 0 && video.currentTime < 0) {
+    video.currentTime = 0
+  } else if (timeDelta > 0 && video.currentTime > video.duration) {
+    video.currentTime = video.duration
+  } else {
```

Comment:
Shouldn't this be as follows?
```suggestion
  if (timeDelta < 0 && (video.currentTime + timeDelta) < 0) {
    video.currentTime = 0
  } else if (timeDelta > 0 && (video.currentTime + timeDelta) > video.duration) {
    video.currentTime = video.duration
  } else {
```

Reply:
Yes! Thanks 👍